### PR TITLE
Move citus internal functions

### PR DESCRIPTION
**Pull request by eaydingol**
Move more functions to citus_internal schema, the list:

citus_internal_add_placement_metadata
citus_internal_add_shard_metadata
citus_internal_add_tenant_schema
citus_internal_adjust_local_clock_to_remote
citus_internal_database_command

#7405

### CODE REVIEW

File: src/backend/distributed/metadata/metadata_sync.c [hunk 1]
```diff
@@ -4056,7 +4056,7 @@ citus_internal_database_command(PG_FUNCTION_ARGS)
 	}
 	else
 	{
-		ereport(ERROR, (errmsg("citus_internal_database_command() can only be used "
+		ereport(ERROR, (errmsg("citus_internal.database_command() can only be used "
```
Comment (eaydingol) [inline src/backend/distributed/metadata/metadata_sync.c:4059]: The line is covered, see [src/test/regress/expected/create_drop_database_propagation.out](https://github.com/citusdata/citus/pull/7470/files#diff-51744aa2356f2516b6e5ae92e2e1e804a67bed8e55df6b3fb607d0f4a0bab52e) line 27-28

File: src/backend/distributed/sql/udfs/citus_internal_add_placement_metadata/latest.sql [hunk 2]
```diff
@@ -1,3 +1,27 @@
...
+CREATE OR REPLACE FUNCTION citus_internal.add_placement_metadata(
+							shard_id bigint, shard_state integer,
+							shard_length bigint, group_id integer,
+							placement_id bigint)
+    RETURNS void
+    LANGUAGE C STRICT
+    AS 'MODULE_PATHNAME', $$citus_internal_add_placement_metadata_legacy$$;
+
+COMMENT ON FUNCTION citus_internal.add_placement_metadata(bigint, integer, bigint, integer, bigint) IS
+    'Inserts into pg_dist_shard_placement with user checks';
```
Comment (JelteF) [inline src/backend/distributed/sql/udfs/citus_internal_add_placement_metadata/latest.sql:23]: I believe this one is not necessary. The definition you copied this from is also a legacy one to support rolling upgrades. We're not calling this 5 argument version in our codebase anymore.

**Review by JelteF — APPROVED** [review]
Looks good, left one comment. Feel free to merge after addressing that one.

File: src/backend/distributed/metadata/metadata_sync.c (hunk 1 above)
Comment (JelteF) [reply src/backend/distributed/metadata/metadata_sync.c:4059]: codecov has been a bit unreliable recently, reporting things as uncovered even though they are. 

In either case, this isn't really a new line so no need to add coverage for it in this PR.

File: src/backend/distributed/sql/udfs/citus_internal_add_placement_metadata/latest.sql (hunk 2 above)
Comment (eaydingol) [reply src/backend/distributed/sql/udfs/citus_internal_add_placement_metadata/latest.sql]: It is removed in the last commit and a test case is updated.

**Pull request merged**